packages/wasm-dpp/test/unit/stateTransition/StateTransitionFactory.spec.js

```diff
@@ -1,160 +1,99 @@
+    stateRepositoryMock.fetchDocuments.resolves();
+    stateRepositoryMock.createDocument.resolves();
+    stateRepositoryMock.updateDocument.resolves();
+    stateRepositoryMock.removeDocument.resolves();
+    stateRepositoryMock.fetchTransaction.resolves();
+    stateRepositoryMock.fetchIdentity.resolves();
+    stateRepositoryMock.createIdentity.resolves();
+    stateRepositoryMock.addKeysToIdentity.resolves();
+    stateRepositoryMock.disableIdentityKeys.resolves();
+    stateRepositoryMock.updateIdentityRevision.resolves();
+    stateRepositoryMock.addToIdentityBalance.resolves();
+    stateRepositoryMock.fetchIdentityBalance.resolves();
+    stateRepositoryMock.fetchIdentityBalanceWithDebt.resolves();
+    stateRepositoryMock.addToSystemCredits.resolves();
+    stateRepositoryMock.fetchLatestPlatformBlockHeight.resolves();
+    stateRepositoryMock.fetchLatestPlatformCoreChainLockedHeight.resolves();
+    stateRepositoryMock.verifyInstantLock.resolves();
+    stateRepositoryMock.markAssetLockTransactionOutPointAsUsed.resolves();
+    stateRepositoryMock.verifyChainLockHeight.resolves();
+    stateRepositoryMock.isAssetLockTransactionOutPointAlreadyUsed.resolves();
+    stateRepositoryMock.fetchSMLStore.resolves();
+    stateRepositoryMock.fetchLatestWithdrawalTransactionIndex.resolves();
+    stateRepositoryMock.enqueueWithdrawalTransaction.resolves();
+    stateRepositoryMock.fetchLatestPlatformBlockTime.resolves();
+
+    const blsAdapter = await getBlsAdapterMock();
 
     factory = new StateTransitionFactory(
-      validateStateTransitionBasicMock,
-      createStateTransitionMock,
-      dppMock,
-      decodeProtocolEntityMock,
+      stateRepositoryMock,
+      blsAdapter,
     );
   });
 
   describe('createFromObject', () => {
     it('should return new State Transition with data from passed object', async () => {
-      validateStateTransitionBasicMock.returns(new ValidationResult());
-
       const result = await factory.createFromObject(rawStateTransition);
 
-      expect(result).to.equal(stateTransition);
-
-      expect(validateStateTransitionBasicMock).to.have.been.calledOnceWith(rawStateTransition);
-
-      expect(createStateTransitionMock).to.have.been.calledOnceWith(rawStateTransition);
+      expect(result.toObject()).to.deep.equal(stateTransition.toObject());
     });
 
     it('should return new State Transition without validation if "skipValidation" option is passed', async () => {
       const result = await factory.createFromObject(rawStateTransition, { skipValidation: true });
 
-      expect(result).to.equal(stateTransition);
-
-      expect(validateStateTransitionBasicMock).to.have.not.been.called();
-
-      expect(createStateTransitionMock).to.have.been.calledOnceWith(rawStateTransition);
-    });
-
-    it('should throw InvalidStateTransitionError if passed object is not valid', async () => {
-      const validationError = new SomeConsensusError('test');
```

Comment:
Can you just create a test where `InvalidStateTransitionError` is thrown? Without any mocks and stuff

Reply:
I'll try